spec/unit/git/rugged/credentials_spec.rb

```diff
@@ -79,6 +79,35 @@
     end
   end
 
+  describe "generating github app tokens" do
+    it 'errors if app id has invalid characters' do
+      expect { subject.github_app_token("123A567890", "fake", "300")
+      }.to raise_error(R10K::Git::GitError, /App id contains invalid characters/)
+    end
+    it 'errors if app ttl has invalid characters' do
+      expect { subject.github_app_token("123456", "fake", "abc")
+      }.to raise_error(R10K::Git::GitError, /Github App token ttl contains/)
+    end
+    it 'errors if private file does not exist' do
+      R10K::Git.settings[:github_app_key] = "/missing/token/file"
+      expect(File).to receive(:readable?).with("/missing/token/file").and_return false
```

Comment:
If we're setting this, might as well use it everywhere ¯\_(ツ)_/¯ 
```suggestion
      expect(File).to receive(:readable?).with(R10K::Git.settings[:github_app_key]).and_return false
```